### ISSUE TITLE
Feature request: training plans

### ISSUE THREAD

**Issue by bernerbrau** [jonasoreland/runnerup#314]
Before I used RunnerUp I used the ZenLabs training apps (C25k, 10k trainer) with predetermined workout plans. I switched to RunnerUp because of the ability to better customize my workouts, but the one thing I miss is the ability to queue up a whole bunch of workouts of increasing difficulty over several weeks to several months. 

Here's how I envision it at a high level: Leverage the existing workout data (Basic/Interval/Advanced) to create plans by parameterizing/overriding certain values, like distance/time/target pace, etc. Then you would just select a workout for each day of the week, enter in your goal distance/time/pace, the number of weeks you have until your goal/race/etc, and the app would calculate a linear progression from your first week to your last. You could then see the workouts laid out in a calendar, and customize them at will using the existing workout UI. Then when it's time to work out, you click the "plans" tab, it queues up your next workout, and you hit Start. 

Thoughts? Interest?

**Comment by nebmo** [comment]
Sounds like a good idea. But it is not easy. Training programs are highly individual. But with parameters like distance, target time and number of exercises per week and by looking at existing training programs (I guess there are multiple websites that has this) it could be doable..

**Comment by bernerbrau** [comment]
Yeah - I'll admit building a whole training program out of a few sample workouts and some start and end parameters is pretty limited. Using existing training programs as a template is a good idea - as long we can get permission to use them.

**Comment by CharString** [comment]
Currently there is a non-free way. It *is* gratis, but uses a closed source software service.

With a Garmin Connect account, you can create workouts and plan them in your calendar, or use one of the many pre-made training plans that Garmin provides.  The workout(s) you have planned for today will be marked with a * in the manage workout dialog in RunnerUp, where you can download it to your phone.

You don't have to upload your tracks to Garmin, in order to be able to use this part of their service.

**Comment by davidedelvento** [comment]
I am also very interested in this. As a first step, how we make the following change to the existing advanced workout (json)?

Allow one to not specify a numeric value for pace/distance etc, but use symbols instead? For example, one would create a json like:

- warmup for `x` minutes at `w` speed
- repeat `n` times
  - run `y` minutes at `fast` speed
  - run `z` minutes at `recovery` speed

Elsewhere in the interface one would define what these `x`, `y`, `z`, `n`, `w`, `fast`, `recovery` values are. If one creates a bunch of such json (like some advanced people do), it would be easy to change these values instead of the individual json.

At a second stage, these values could be used for a progressively increasing training program.

I'll be happy to implement the first stage, if you think this is a good idea
